etcd/v3/kv_etcd.go

```diff
@@ -733,7 +733,10 @@ func (et *etcdKV) LockWithTimeout(
 			}
 		}
 		if err != nil && time.Since(startTime) > lockTryDuration {
-			return nil, err
+			currLockerTag := ec.LockerIDInfo{LockerID: ""}
+			if _, errGet := et.GetVal(key, &currLockerTag); errGet == nil {
+				return nil, fmt.Errorf("failed to take a lock on %v: lock taken by: %v ", key, currLockerTag.LockerID)
```

Comment:
We should still return an error here if errGet != nil

Reply:
```suggestion
			if _, errGet := et.GetVal(key, &currLockerTag); errGet == nil {
				return nil, fmt.Errorf("failed to take a lock on %v: lock taken by: %v ", key, currLockerTag.LockerID)
			}
			return nil, fmt.Errorf("failed to take a lock on %v", key)
```

Reply:
Okay, I'll raise a new PR for this one because this one has been closed